### PR TITLE
Prevents headless mobs from being flashed

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
This will basically only ever apply to changelings and dullahans

### CODE REVIEW

**Comment by Thunder12345** [comment]
Dullahans: The ultimate rev meta

**Comment by Cyberboss** [comment]
Shouldn't the eyes check already handle this?

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -279,6 +279,10 @@
 		if(visual)
 			return
 
+		var/obj/item/bodypart/head/head = H.get_bodypart("head")
+		if(!head)
```
Comment (lzimann) [inline code/modules/mob/living/carbon/carbon_defense.dm]: just `if(!H.get_bodypart("head"))` will do here, no need to declare the var

**Comment by KorPhaeron** [comment]
@Cyberboss If you lop off a lings head it will get new eyes when it revives despite still missing a head

eyes

**Comment by QualityVan** [comment]
Yeah, changelings don't regenerate heads by default when they undeath themselves. Kept that way for humorous purposes. They get all their head organs back, though, and if they activate regeneration normally(not the death stasis version) they'll get the bodypart back. I guess dullahans have eyes too?

**Comment by KorPhaeron** [comment]
Yeah

I guess alternative solutions would be to give abstract eyes flash protect, or some kind of flag for carbons that they dont need eyes to see

**Comment by QualityVan** [comment]
Changelings do need eyes to see(and are susceptible to flashing, to my knowledge), they just don't need a head to keep their eyes in. Flash protection would work for dullahans though if they have specific eyes.

**Comment by DoctoreMad** [comment]
You should totally be able to flash a dullahan's head to stun the body or... something.
It makes sense to me.

**Comment by DoctoreMad** [comment]
Also how does this fix work with pepper spray?

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 2]
```diff
@@ -279,6 +279,9 @@
 		if(visual)
 			return
 
+	if(!get_bodypart("head"))
+		return
```
Comment (PKPenguin321) [inline code/modules/mob/living/carbon/carbon_defense.dm]: put this right after the parent call instead of down here to fix travis

**Comment by PKPenguin321** [comment]
this fix does not touch pepper spray

**Comment by PKPenguin321** [comment]
Shouldn't be able to flash headless changelings even if they do still technically have eye organs, because think about it, if you see a headless dude and want to flash a light in his eyes, where are you gonna flash him? You don't know, and as far as you can tell there isn't a way

**Comment by DoctoreMad** [comment]
You should totally see that I wrote down Dullahan and not Changeling so I don't have to waste a response waiting you to make a new argument!

**Comment by QualityVan** [comment]
PKP, you don't have to aim a flash at someone's eyes right now. You can flash at their torso and it'll work fine.

**Pull request closed without merging**